galsim/phase_psf.py

```diff
@@ -827,14 +827,35 @@ def _reset(self):
                 pass
         self._update_attrs()
 
-    def instantiate(self, _bar=None, **kwargs):
-        for layer in self:
-            try:
-                layer.instantiate(**kwargs)
-            except AttributeError:
-                pass
-            if _bar:  # pragma: no cover
-                _bar.update()
+    def instantiate(self, pool=None, _bar=None, **kwargs):
+        """Instantiate the screens in this PhaseScreenList.
+
+        @param pool      A multiprocessing.Pool object to use to instantiate screens in parallel.
+        @param **kwargs  Keyword arguments to forward to screen.instantiate().
+        """
+        if pool is not None:
+            import sys
+            assert_str = "instantiating PhaseScreenList with pool requires python version >= 3.4"
+            assert sys.version_info >= (3,4), assert_str
+
+            results = []
+            for layer in self:
+                try:
+                    results.append(pool.apply_async(layer.instantiate, kwds=kwargs))
```

Comment:
I think this function is available in Python 2.7.  https://docs.python.org/2/library/multiprocessing.html#module-multiprocessing.pool

What part of this requires python 3.4?

Reply:
This line doesn't require py3.4 per se, but I think using the downstream layer.instantiate in a pool but without the shared memory stuff will effectively fail.  I.e., it will create the screens in the subprocesses, but not propagate them back to the parent since w/o py3.4 it's not using shared memory.  (At least, I think that's what would happen, I haven't actually tried it).

I'll try out your suggestion to use the py2.7 versions of RawArray, etc though, in which case I agree this assert should be removed.

Reply:
This now works on py2.7 too.